### ISSUE TITLE
Warning: Not implemented: process.dlopen doesn't support 'flags' argument

### ISSUE THREAD

**Issue by janpio** [denoland/deno#20075]
A user reported that when using Prisma with Deno 1.36, you get the following warning right now:
```
deno run -A main.ts
Uncaught Error: Assertion failed
Warning: Not implemented: process.dlopen doesn't support 'flags' argument
[
  { id: 1, email: "test@test.com", name: null },
  { id: 2, email: "test2@test.com", name: null },
  { id: 3, email: "test3@test.com", name: null }
]
```
https://github.com/prisma/prisma/issues/2452#issuecomment-1666513809

Seems https://github.com/denoland/deno/pull/19860 implemented support for `process.dlopen`, but not the `flags` argument.

Prisma indeed uses `flags`: https://github.com/prisma/prisma/blob/3276015cee8358b3706b5e3014cbafae7b8eb529/packages/client/src/runtime/core/engines/library/DefaultLibraryLoader.ts#L39-L57C56

**Comment by bartlomieju** [comment]
@janpio would you be fine if we just ignored this field altogether? It appears the library we use for opening N-API files doesn't support "deepbind" flag anyway: https://docs.rs/libloading/latest/libloading/os/unix/index.html

**Comment by janpio** [comment]
I don't have the slightest idea 🤷 😆 

I see that it was introduced via https://github.com/prisma/prisma/pull/18426, and before we basically used a lazy `require` - but had to switch to `dlopen` with those flag to fix some _very_ nasty bugs on our side related to OpenSSL. Maybe @aqrln knows more.

**Comment by bartlomieju** [comment]
Sounds good, I'll open a PR that removes the warning tonight and it will be released next week in v1.36.2.

**Comment by aqrln** [comment]
> As far as I can tell, Deno only exports napi symbols

Actually, looking at your [`build.rs`](https://github.com/denoland/deno/blob/91dc6fa5f11f5621c21397c783ee899bfd9657ef/cli/build.rs#L431-L443) and not just on `nm -D $(which deno)` output on my system, it seems that it's only true when building on a system with modern glibc, otherwise it exports everything.

Looking at the dynamic symbols table in the deno binary, the only thing with global binding and default visibility that strikes my attention is a bunch of sqlite symbols, but those shouldn't pose a problem even when exported since we statically link sqlite.
